generator/graphql-kotlin-federation/src/main/kotlin/com/expediagroup/graphql/generator/federation/extensions/CompletableFutureExtensions.kt

```diff
@@ -18,12 +18,42 @@ package com.expediagroup.graphql.generator.federation.extensions
 import java.util.concurrent.CompletableFuture
 
 /**
- * Returns a new CompletableFuture of a list with the resolved values of the given CompletableFutures.
- * the returned completableFuture will complete when all given CompletableFutures complete.
+ * Returns a [CompletableFuture] that completes when all the input futures have completed,
+ * with a list of the resolved values obtained from the completed futures.
+ * If any of the input futures complete exceptionally, then the returned [CompletableFuture] also completes exceptionally
+ * with a [java.util.concurrent.CompletionException] holding the exception as its cause.
+ *
+ * @return a [CompletableFuture] that completes with a list of resolved values.
  */
-internal fun <T : Any?> List<CompletableFuture<out T>>.collectAll(): CompletableFuture<List<T>> =
+internal fun <T : Any?> List<CompletableFuture<out T>>.joinAll(): CompletableFuture<List<T>> =
     CompletableFuture.allOf(
         *toTypedArray()
     ).thenApply {
         map(CompletableFuture<out T>::join)
     }
+
+/**
+ * Returns a [CompletableFuture] that completes when all the input futures have completed,
+ * with a list of [Result] objects that indicate whether each future completed successfully or with an error.
+ * If a future completed with an error, the corresponding [Result] object will contain the exception that was thrown.
+ *
+ * @return a [CompletableFuture] that completes with a list of [Result] objects.
+ */
+@Suppress("TooGenericExceptionCaught")
+internal fun <T : Any?> List<CompletableFuture<out T>>.allSettled(): CompletableFuture<List<Result<T>>> {
```

Comment:
instead of `allSettled()` maybe this should be `safelyCompleteAll()` or something along those lines

Reply:
had the same doubt about how to name this extension fn, then saw this
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/allSettled